core/src/main/java/org/elasticsearch/action/delete/DeleteRequest.java

```diff
@@ -241,6 +250,7 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeString(type);
         out.writeString(id);
         out.writeOptionalString(routing());
+        out.writeOptionalString(parent());
```

Comment:
similarly to the read, you need to protect the write with a version check. Additionally since you changed the parent setter to not set the routing, I think you would need to do the following:

```java
if (out.getVersion().onOrAfter(Version.V_2_2_0)) {
  out.writeOptionalString(routing());
  out.writeOptionalString(parent());
} else {
  String routing = routing();
  if (routing == null) {
    routing = parent();
  }
  out.writeOptionalString(routing);
}
```